assets/src/edit-story/components/checklist/checks/storyAmpValidationErrors.js

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { trackEvent } from '@web-stories-wp/tracking';
+import { useEffect, useRef, useState } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { useStory } from '../../../app';
+import { ChecklistCard, DefaultFooterText } from '../../checklistCard';
+import { PRIORITY_COPY } from '../constants';
+import { useRegisterCheck } from '../countContext';
+
+export async function getStoryAmpValidationErrors({ link, status }) {
+  // TODO: Enable for drafts as well?
+  if (!link || !['publish', 'future'].includes(status)) {
+    return false;
+  }
+
+  try {
+    const response = await fetch(link);
+    const storyMarkup = await response.text();
+    const { status: markupStatus, errors } =
+      window.amp.validator.validateString(storyMarkup);
+
+    if ('FAIL' !== markupStatus) {
+      return false;
+    }
+
+    const filteredErrors = errors
+      .filter(({ severity }) => severity === 'ERROR')
```

Comment:
these strings are only used here in this filtering, didn't feel necessary to pull them out as constants - lemme know if you disagree and i'll update.